src/main/java/github/scarsz/discordsrv/modules/alerts/AlertListener.java

```diff
@@ -0,0 +1,281 @@
+            String triggerEvent = alert.get("Trigger").convert().intoString();
+
+            // make sure the called event matches what this alert is supposed to trigger on
+            if (!event.getEventName().equalsIgnoreCase(triggerEvent)) return;
+
+            Player player = event instanceof PlayerEvent ? ((PlayerEvent) event).getPlayer() : null;
+
+            // make sure channel is available
+            String gameChannel = alert.get("Channel").asString();
+            if (gameChannel == null) {
+                DiscordSRV.debug("Not running alert for event " + triggerEvent + ": no target channel was defined");
+                return;
+            }
+            TextChannel channel = DiscordSRV.getPlugin().getDestinationTextChannelForGameChannelName(gameChannel);
+            if (channel == null) {
+                DiscordSRV.debug("Not running alert for event " + triggerEvent + ": target TextChannel was not available");
+                return;
+            }
+
+            // make sure alert should run even if event is cancelled
+            if (event instanceof Cancellable && ((Cancellable) event).isCancelled()) {
+                Boolean ignoreCancelled = alert.get("IgnoreCancelled").as(Boolean.class);
+                if (ignoreCancelled == null || ignoreCancelled) {
+                    DiscordSRV.debug("Not running alert for event " + triggerEvent + ": event was cancelled");
+                    return;
+                }
+            }
+
+            // check alert conditions
+            boolean anyConditionFailed = alert.dget("Conditions").children()
+                    .anyMatch(dynamic -> {
+                        String expression = dynamic.convert().intoString();
+                        Boolean value = new SpELExpressionBuilder(expression)
+                                .withPluginVariables()
+                                .withVariable("event", event)
+                                .withVariable("server", Bukkit.getServer())
+                                .withVariable("discordsrv", DiscordSRV.getPlugin())
+                                .withVariable("player", player)
+                                .withVariable("channel", channel)
+                                .withVariable("jda", DiscordUtil.getJda())
+                                .evaluate(event, Boolean.class);
+                        DiscordSRV.debug("Condition \"" + expression + "\" -> " + value);
+                        return !(value != null ? value : false);
+                    });
+            if (anyConditionFailed) return;
+
+            // construct the alert message
+            MessageFormat messageFormat = new MessageFormat();
+
+            Dynamic contentDynamic = alert.dget("Content");
+            if (contentDynamic.isPresent() && StringUtils.isNotBlank(contentDynamic.asString())) messageFormat.setContent(contentDynamic.asString());
+
+            if (alert.dget("Webhook").isPresent()) {
```

Comment:
Reason not to use `DiscordSRV#getMessageFromConfiguration`?

Reply:
Because that takes a base key, don’t think it’s usable with a list like this. Was going to refactor it to take a root dynamic instead

Reply:
[`2b5a36e`](https://github.com/DiscordSRV/DiscordSRV/pull/759/commits/2b5a36e506a7ebf164c9beb05644ab0c1eae2c78)